### PR TITLE
ci: use crates api to get owner list

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

seems that we can't get CRATE_TOKEN correctly in a PR.

#### Summary of Changes

just use curl to get crates owner list

Screenshot 2023-04-26 at 8 18 18 AM

### CODE REVIEW

**Review by CriesofCarrots — APPROVED** [review]
:shipit:

File: ci/check-crates.sh [hunk 1]
```diff
@@ -50,9 +50,11 @@ for file in "${files[@]}"; do
     fi
   done
 
-  result="$(cargo owner --list -q "$crate_name" --token "$CRATE_TOKEN" 2>&1)"
-  if [[ $result =~ ^error ]]; then
-    if [[ $result == *"Not Found"* ]]; then
+  response=$(curl -s https://crates.io/api/v1/crates/"$crate_name"/owners)
```
Comment (t-nelson) [inline ci/check-crates.sh:53]: i think we'll be subject to lower api rate-limits without the token, so we need to keep an eye on that

**Comment by ryoqun** [comment]
> seems that we can't get CRATE_TOKEN correctly in a PR.

(hi, i'm becoming a github actions/workflow pro recently) well, this is expected. pr can't access to secretes as indicated below:

image

**Comment by yihau** [comment]
yeah! thank you for the useful checking method! 

I think they changed something (like fix or publish new features) cuz our PR works fine with the same method earlier. (like this one https://github.com/solana-labs/solana/actions/runs/4778963531/jobs/8495728916 ) and iirc when I implement this feature, we can still get the secret correctly if we don't modify the yml. (sounds risky today though) it just break on that day  🤔

**Pull request merged**